### PR TITLE
Convert ReactImagePropertyList to kotlin

### PR DESCRIPTION
**Pull request by bufgix**
## Summary:
Issue: https://github.com/facebook/react-native/issues/38825#issuecomment-1669205837

## Changelog:
[GENERAL] [CHANGED] - ReactImagePropertyList.java => ReactImagePropertyList.kt

## Test Plan:
`./gradlew :packages:react-native:ReactAndroid:test`

### CODE REVIEW

**Comment by cortinico** [comment]
Hey @bufgix 
Could you accept our CLA? We can't merge code from contributors which don't accept it otherwise.

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/views/image/ReactImagePropertyTest.kt [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+/** Verify that [ScalingUtils] properties are being applied correctly by [ ]. */
+@PrepareForTest(Arguments::class, RNLog::class)
+@RunWith(RobolectricTestRunner::class)
+@PowerMockIgnore("org.mockito.*", "org.robolectric.*", "androidx.*", "android.*")
+class ReactImagePropertyTes {
+
+    @get:Rule var rule = PowerMockRule()
+
+    private var mContext: ReactApplicationContext? = null
```
Comment (mdvacca) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/views/image/ReactImagePropertyTest.kt:45]: let's replace mContext -> context

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/views/image/ReactImagePropertyTest.kt [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    private var mCatalystInstanceMock: CatalystInstance? = null
```
Comment (mdvacca) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/views/image/ReactImagePropertyTest.kt:46]: let's replace mCatalystInstanceMock -> catalystInstanceMock

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/views/image/ReactImagePropertyTest.kt [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    private var mThemeContext: ThemedReactContext? = null
```
Comment (mdvacca) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/views/image/ReactImagePropertyTest.kt:47]: NIT: let's replace mThemeContext -> themeContext

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/views/image/ReactImagePropertyTest.kt [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        Fresco.initialize(mContext)
+        DisplayMetricsHolder.setWindowDisplayMetrics(DisplayMetrics())
+    }
+
+    @After
+    fun teardown() {
+        DisplayMetricsHolder.setWindowDisplayMetrics(null)
+    }
+
+    fun buildStyles(vararg keysAndValues: Any?): ReactStylesDiffMap {
```
Comment (mdvacca) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/views/image/ReactImagePropertyTest.kt:71]: NIT: let's make this method private

**Review by mdvacca — COMMENTED** [review]
Thanks for working on this PR, added just few small comments

**Review by cortinico — APPROVED** [review]

**Pull request closed without merging**